library/core/src/panicking.rs

```diff
@@ -92,6 +93,17 @@ pub fn panic_fmt(fmt: fmt::Arguments<'_>) -> ! {
     unsafe { panic_impl(&pi) }
 }
 
+/// This function is used instead of panic_fmt in const eval.
+#[cfg(not(bootstrap))]
+#[lang = "const_panic_fmt"]
+pub const fn const_panic_fmt(fmt: fmt::Arguments<'_>) -> ! {
+    if let Some(msg) = fmt.as_str() {
+        panic_str(msg);
+    } else {
+        panic_str("???");
```

Comment:
If this is unreachable today, could the `???` be replaced with a more descriptive message, or maybe just an `unreachable_unchecked` call (since I'm guessing miri would be able to handle that).